diff --git a/lib/asset_packager.js b/lib/asset_packager.js
index a45ab07..05fdbef 100644
--- a/lib/asset_packager.js
+++ b/lib/asset_packager.js
@@ -132,7 +132,7 @@ AssetPackager.prototype.getSourceAsset = function(assetPath, done) {
           analyzer(asset, self, next);
         }
       ], function(err) {
-        if (self.watch && !self._watching[assetPath]) {
+        if (!err && self.watch && !self._watching[assetPath]) {
           self._watching[assetPath] = 
             FS.watch(assetPath, { persistant: false }, self.invalidate);
         }
diff --git a/lib/utils.js b/lib/utils.js
index 237a4e3..d4618a1 100644
--- a/lib/utils.js
+++ b/lib/utils.js
@@ -95,7 +95,11 @@ function _mkdir_p(path, done) {
         making[path] = ASYNC.memoize(function(done) {
           _mkdir_p(PATH.dirname(path), function(err) {
             if (err) return done(err);
-            FS.mkdir(path, done);
+            FS.mkdir(path, function(err) {
+              // sometimes another routine gets around to making this first
+              if (err && err.code === 'EEXIST') err = null;
+              done(err);
+            });
             delete making[path];
           });
         });